homeassistant/components/shelly/entity.py

```diff
@@ -352,9 +356,13 @@ def __init__(self, coordinator: ShellyRpcCoordinator, key: str) -> None:
         """Initialize Shelly entity."""
         super().__init__(coordinator)
         self.key = key
-        self._attr_device_info = {
-            "connections": {(CONNECTION_NETWORK_MAC, coordinator.mac)}
-        }
+
+        if blu_addr := coordinator.device.config.get(key, {}).get("addr"):
```

Comment:
Cannot we avoid the if like we did in the main PR for BLU TRV ?

Reply:
We don’t know here if the entity comes from BLU TRV or not. Or maybe I don’t get your idea.

Reply:
As we blocked another PR with a `if` in entity.py, I was just looking for a different way of approaching.

Reply:
Which PR will block this? And why?